BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -154,30 +200,116 @@ extension ElementContent.Builder: ContentStorage {
         return result
     }
 
-    private var layoutItems: [(LayoutType.Traits, Measurable)] {
-        return children.map { ($0.traits, $0) }
+    private func layoutItems(in environment: Environment) -> [(LayoutType.Traits, Measurable)] {
+        return children.map { ($0.traits, $0.content.measurable(in: environment)) }
     }
 
 }
 
 
 extension ElementContent.Builder {
 
-    fileprivate struct Child: Measurable {
+    fileprivate struct Child {
 
         var traits: LayoutType.Traits
         var key: String?
         var content: ElementContent
         var element: Element
 
-        func measure(in constraint: SizeConstraint) -> CGSize {
-            return content.measure(in: constraint)
-        }
+    }
+
+}
 
+private struct EnvironmentAdaptingStorage: ContentStorage {
```

Comment:
If we're going to make envs a core part of blueprint, I would prefer we pay the tax of moving this stuff onto ContentStorage itself, I think.

Reply:
Do you mean moving the environment adapter onto `ElementContent.Builder` (the existing `ContentStorage`)? I actually had it that way in an earlier version, and moved it onto this new type to clean things up.